src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -36,15 +40,16 @@ const SearchPage = () => {
   `;
 
   const [searchText, setSearchText] = useState('');
-  const [results, setResults] = useState([]);
+  const [results, setResults] = useState(undefined);
+  const [filter, setFilter] = useState('author');
 
-  const [executeSearch, { data }] = useLazyQuery(SEARCH_QUERY, {
-    // setting author variable here to use in the query above
+  const [executeSearch, { loading, data }] = useLazyQuery(SEARCH_QUERY, {
+    // Setting author variable here to use in the query above
     variables: { author: searchText },
-
     onCompleted: () => {
       // We only want to display a feed once, this would avoid the issue if
       // there are multiple posts by the same author returned in our query
+      setResults(undefined);
```

Comment:
I wonder if this is what is causing the brief "No results" text to flash before we see the actual results.  Why do you bother with this?

Reply:
Removed, I think it was left there when I was trying to find ways to get the spinner to render

Reply:
Ah, also if you're trying to search Posts, the route we're using will only work when searching `strings.length > 3`. This should explain why you're seeing the no results as you're typing.